src/transformers/models/mobilevit/modeling_mobilevit.py

```diff
@@ -0,0 +1,1083 @@
+            out_channels=config.neck_hidden_sizes[1],
+            stride=1,
+            num_stages=1,
+        )
+        self.layer.append(layer_1)
+
+        layer_2 = self._make_mobilenet_layer(
+            in_channels=config.neck_hidden_sizes[1],
+            out_channels=config.neck_hidden_sizes[2],
+            stride=2,
+            num_stages=3,
+        )
+        self.layer.append(layer_2)
+
+        layer_3 = self._make_mobilevit_layer(
+            in_channels=config.neck_hidden_sizes[2],
+            out_channels=config.neck_hidden_sizes[3],
+            stride=2,
+            hidden_size=config.hidden_sizes[0],
+            num_stages=2,
+        )
+        self.layer.append(layer_3)
+
+        if dilate_layer_4:
+            dilation *= 2
+
+        layer_4 = self._make_mobilevit_layer(
+            in_channels=config.neck_hidden_sizes[3],
+            out_channels=config.neck_hidden_sizes[4],
+            stride=2,
+            hidden_size=config.hidden_sizes[1],
+            num_stages=4,
+            dilation=dilation,
+        )
+        self.layer.append(layer_4)
+
+        if dilate_layer_5:
+            dilation *= 2
+
+        layer_5 = self._make_mobilevit_layer(
+            in_channels=config.neck_hidden_sizes[4],
+            out_channels=config.neck_hidden_sizes[5],
+            stride=2,
+            hidden_size=config.hidden_sizes[2],
+            num_stages=3,
+            dilation=dilation,
+        )
+        self.layer.append(layer_5)
+
+    def _make_mobilenet_layer(
+        self,
+        in_channels: int,
+        out_channels: int,
+        stride: int,
+        num_stages: int,
+    ) -> nn.Sequential:
+        block = []
+        for i in range(num_stages):
+            layer = MobileViTInvertedResidual(
+                self.config,
+                in_channels=in_channels,
+                out_channels=out_channels,
+                stride=stride if i == 0 else 1,
+            )
+            block.append(layer)
+            in_channels = out_channels
+
+        return nn.Sequential(*block)
+
+    def _make_mobilevit_layer(
+        self,
+        in_channels: int,
+        out_channels: int,
+        stride: int,
+        hidden_size: int,
+        num_stages: int,
+        dilation: int = 1,
+    ) -> nn.Sequential:
+        block = []
+        if stride == 2:
+            layer = MobileViTInvertedResidual(
+                self.config,
+                in_channels=in_channels,
+                out_channels=out_channels,
+                stride=stride if dilation == 1 else 1,
+                dilation=dilation // 2 if dilation > 1 else 1,
+            )
+            block.append(layer)
+            in_channels = out_channels
+
+        layer = MobileViTLayer(
+            self.config,
+            in_channels=in_channels,
+            hidden_size=hidden_size,
+            num_stages=num_stages,
+        )
+        block.append(layer)
+
+        return nn.Sequential(*block)
```

Comment:
```suggestion
        return nn.ModuleList(*block)
```
Same here

Reply:
Can't do that here since ModuleList doesn't have a forward method.